architecting-cloud-native-dotnet-apps-for-azure/observability-patterns.md

```diff
@@ -3,12 +3,57 @@ title: Observability patterns
+* Information
+* Warning
+* Error
+* Fatal
+
+These many log levels provide granularity in logging. When the application is functioning properly in production it may be configured to only log importing messages. When the application is misbehaving then the log level can be increased so more verbose logs are gathered.  This balances performance and debuggability. 
+
+The high performance of logging tools and the tunability of verbosity should encourage developers to log as frequently as possible. Many favour a pattern of logging entry and exit from each method. This may sound like overkill but it is very infrequent that developers will wish for less logging. In fact it is not uncommon to perform deployments for the sole purpose of adding logging around a particularly problematic method. Err on the side of too much logging and not too little. 
+
+In traditional applications the log files were typically stored on the local machine. in fact, on Unix-like operating systems there is a folder structure defined, typically under `/var/log`, to hold any logs. Of course, the utility of logging to a flat file on a single machine is vastly reduced in a cloud environment. Applications producing logs may not have access to the local disk or the local disk may be highly transient as containers are shuffled around physical machines.
+
+Cloud applications developed with an eye to microservices also pose some challenges for the file system based loggers of yesteryear. A user interaction may now span multiple services which run on different machines. It would be very challenging to correlate the logs from a user or a session. 
+
+Finally, the number of users in some cloud applications is quite high. Imagine that each user generates a hundred lines of log when the login to an application. In isolation that is quite manageable but multiply that over a hundred thousand user and the volume of logs becomes challenging. 
+
+![Logs from various sources are ingested into a centralized monitoring tool.](media/centralizedLogging.png)
+ 
+Fortunately, there are some fantastic alternatives to using file system logs. A centralized log server to which all logs are sent fixes all the problems. Logs are collected by the applications and shipped to a central logging application which indexes and stores the logs. This class of system is capable of ingesting tens of gigabytes of logs every day. 
+
+It is also helpful to follow some standard practices when building logging which runs across many services. For instance, generating a correlation Id at the start of a length interaction and then logging it in each message related to that interaction makes it easier to search for all related messages. One need only find a single message and extract the correlation Id to find all the related messages. Another example is ensuring that the log format is the same for every service, regardless of the language or logging library it uses. This standardization will make reading logs much easier.
```

Comment:
length interaction -> lengthy
define what a correlation id is, or at least link somewhere that discusses the concept further